File: mobile/lib/features/stay_session/data/mapper/stay_session_mapper.dart
import '../../domain/model/stay_session.dart';
import '../dto/stay_session_response_dto.dart';

class StaySessionMapper {
  const StaySessionMapper();

  StaySession toDomain(StaySessionResponseDto dto) {
    return StaySession(
      id: dto.id ?? '',
      userId: dto.userId ?? '',
      placeName: dto.placeName ?? 'Unknown Place',
      placeType: dto.placeType,
      matchedPlaceSource: dto.matchedPlaceSource,
      startTime: DateTime.tryParse(dto.startTime ?? '') ?? DateTime.now(),
      endTime: DateTime.tryParse(dto.endTime ?? '') ?? DateTime.now(),
      durationMinutes: dto.durationMinutes ?? 0,
      centerLat: dto.centerLat,
      centerLng: dto.centerLng,
      confidence: dto.confidence,
    );
  }
}

File: backend/src/main/java/com/lifeos/backend/staysession/application/StaySessionMapper.java
package com.lifeos.backend.staysession.application;

import com.lifeos.backend.staysession.api.response.StaySessionResponse;
import com.lifeos.backend.staysession.domain.StaySession;
import org.springframework.stereotype.Component;

@Component
public class StaySessionMapper {

    public StaySessionResponse toResponse(StaySession session) {
        return StaySessionResponse.builder()
                .id(session.getId())
                .userId(session.getUserId())
                .startTime(session.getStartTime())
                .endTime(session.getEndTime())
                .durationMinutes(session.getDurationMinutes())
                .placeName(session.getPlaceName())
                .placeType(session.getPlaceType())
                .matchedPlaceSource(session.getMatchedPlaceSource())
                .confidence(session.getConfidence())
                .centerLat(session.getCenterLat())
                .centerLng(session.getCenterLng())
                .build();
    }
}